pydantic/types.py

```diff
@@ -3007,3 +3008,33 @@ def __get_pydantic_core_schema__(cls, source_type: Any, handler: GetCoreSchemaHa
 That is, instead of a [`ValidationError`][pydantic_core.ValidationError] being propagated up and the entire iterable being discarded
 any invalid items are discarded and the valid ones are returned.
 """
+
+
+@_dataclasses.dataclass(**_internal_dataclass.slots_true, frozen=True)
+class FailFast:
+    """
+    A `FailFast` can be used to specify that validation should stop at the first error.
+
+    This can be useful when you want to validate a large amount of data and you only need to know if it's valid or not.
+
+    ```py
+    from typing_extensions import Annotated
+    from pydantic import BaseModel, FailFast
+
+    class Model(BaseModel):
+        x: Annotated[int, FailFast()]
+
+    # This will raise a single error for the first invalid value and stop validation
+    obj = Model(
+        x=[1, 2, 'a', 4, 5, 'b', 7, 8, 9, 10, 'c']
+    )
+    ```
+    """
+
+    fail_fast: bool = True
+
+    def __get_pydantic_core_schema__(self, source_type: Any, handler: GetCoreSchemaHandler) -> CoreSchema:
+        schema = handler(source_type)
+        schema['fail_fast'] = self.fail_fast  # type: ignore
```

Comment:
Not sure it's good solution

Reply:
I think we might be able to skip adding this function here, and instead apply this constraint when we apply other metadata, in `_internal/_known_annotated_metadata.py`

Reply:
Specifically in the `apply_known_metadata` function